### PR TITLE
Add gitlab validations

### PR DESCRIPTION
**Pull request by AradCarmi**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/43580)

## Description
Added validation to the build that fails when there is an open content-test-conf PR with the same branch name as in the content PR which is not merged yet.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .gitlab/ci/on-push.yml [hunk 1]
```diff
@@ -46,6 +46,30 @@ run-validations:
   rules:
     - if: '$NIGHTLY'
 
+validate-content-conf:
+  stage: unittests-and-validations
+  needs: []
+  extends:
+    - .default-job-settings
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/ && $CI_COMMIT_BRANCH !~ /^[0-9]{2}\.[0-9]{1,2}\.[0-9]$/'
```
Comment (dantavori) [inline .gitlab/ci/on-push.yml:55]: please add a comment describing when it runs

**Review by dantavori — APPROVED** [review]
looks good to me 👍🏽 
please attach here examples for builds passing/failing on this step

@esharf please have a look too

**Comment by AradCarmi** [comment]
Pass
Fails

File: .gitlab/ci/on-push.yml [hunk 2]
```diff
@@ -47,6 +47,31 @@ run-validations:
...
     - if: '$NIGHTLY'
 
+# runs in gitlab for the on-push flow, on every new commit pushed to the branch.
+validate-content-conf:
+  stage: unittests-and-validations
+  needs: []
+  extends:
+    - .default-job-settings
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/ && $CI_COMMIT_BRANCH !~ /^[0-9]{2}\.[0-9]{1,2}\.[0-9]$/'
```
Comment (esharf) [inline .gitlab/ci/on-push.yml:57]:
```suggestion
    - if: '$CI_PIPELINE_SOURCE =~ /^(push)$/ && $CI_COMMIT_BRANCH !~ /^[0-9]{2}\.[0-9]{1,2}\.[0-9]$/'
```
this step is not relevant for contributions.

**Review by esharf — APPROVED** [review]

**Review by esharf — APPROVED** [review]

**Pull request merged**